### DISCUSSION TITLE
Changing Facebook App ID and Secret

### DISCUSSION THREAD

**Discussion by goranefbl** [nextauthjs/next-auth#4751] [Help]
### Question 💬

So, I had to change Facebook app, and now whenever one of the old users login with their account, account is not linked anymore and they get "OAuthAccountNotLinked" error. 

I was looking through response and saw that providerAccountId is different between apps, so he is unable to find that account in DB.  This means that nooone from old customers can login anymore. 

It just feels so strange that changing App ID, customers wont be able to login anymore? Anyone had this issue and what would be fix for it? 

### How to reproduce ☕️

Change Facebook APP ID and Secret

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by goranefbl** [comment]
I was thinking of somehow searching via email only and updating providerAccountId with new ones, but not sure where to start.

**Comment by goranefbl** [comment]
So, temporary fix for me was this:

```
async signIn({ user, account, profile, email, credentials }) {
            if (account.provider === "facebook") {
                const userResponse = await prisma.user.findUnique({
                    where: { email: user.email },
                });
                const accountResponse = await prisma.account.findFirst({
                    where: { userId: userResponse.id },
                });
                await prisma.account.update({
                    where: { id: accountResponse.id },
                    data: { providerAccountId: account.providerAccountId },
                });
            }
            return true;
        },
```

So I am updating providerAccountId on every login attempt. Not the best I know, but its working. If anyone knows better solution, do let me know. I would keep this one for a few weeks until active users have logged in enough times.

**Reply by tony-cocco** [reply]
For what it's worth, this code helped me out.

**Comment by balazsorban44** [comment]
From the context of your application, you have created a different provider. You can add multiple providers with different values, as long as they have a unique `id`.

I.e:

```js
...
FacebookProvider({
  clientId: process.env.FACEBOOK1_CLIENT_ID,
  clientSecret: process.env.FACEBOOK1_CLIENT_SECRET
}),
FacebookProvider({
  id: "new-facebook-client", // sign in with `signIn("new-facebook-client")`
  clientId: process.env.FACEBOOK2_CLIENT_ID,
  clientSecret: process.env.FACEBOOK2_CLIENT_SECRET
}),
...
